crates/uv-resolver/src/pubgrub/report.rs

```diff
@@ -1214,6 +1245,26 @@ impl std::fmt::Display for PubGrubHint {
                     "--index-strategy unsafe-best-match".green(),
                 )
             }
+            Self::UnauthorizedIndex { index } => {
+                write!(
+                    f,
+                    "{}{} An index URL ({}) could not be queried due to a lack of valid authentication credentials ({}).",
+                    "hint".bold().cyan(),
+                    ":".bold(),
+                    index.redacted().cyan(),
+                    "401 Unauthorized".bold().red(),
+                )
+            }
+            Self::ForbiddenIndex { index } => {
+                write!(
+                    f,
+                    "{}{} An index URL ({}) could not be queried due to a lack of valid authentication credentials ({}).",
```

Comment:
this is the same as 401, but imo should not be

maybe:
* Index URL could not be queried due to a lack of permissions

or IMO better (because 403s can be, and often are, resource specific)
* Package download for URL (...) was denied by the index.

(do you have access to whole resource URL in this context?)